pkg/rke2/rke2_linux.go

```diff
@@ -199,3 +205,22 @@ func hostnameFQDN() (string, error) {
 
 	return strings.TrimSpace(b.String()), nil
 }
+
+func hostnameFromMetadataEndpoint() (string, error) {
+	resp, err := http.Get("http://169.254.169.254/latest/metadata/local-hostname")
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
```

Comment:
Generic return of bytes is conventionally referred to with a variable named `b`.

```suggestion
	b, err := ioutil.ReadAll(resp.Body)
```

Reply:
Will update.

Reply:
Updated.